### ISSUE TITLE
Size estimation is way off

### ISSUE THREAD

**Issue by aaannndddyyy** [C0D3D3V/Moodle-DL#209]
## Description of the bug
The estimation of total download size is way off

 

## Steps to reproduce the issue
Start moodle-dl and it sync the Moodle files to local drive.
It starts and tells me that the overall size of stuff to be downloaded is ca 70 GB.
I let it run and run, and the size increases and after all night running it says it is 92% complete and has downloaded 173 out of approximately 190 GB. So I let it run just to find it says it has complete 74% out of 214 GB (and 173 out of 214 would be 80 Percent and not 74%, and by tjen it had downloaded more than the 173GB from before.
It's still downloading now, overall size is now 322GB. I wonder if this blows up to infinity. 

## Expected behavior
That it downloads some 70GB as it said or that it says right from the start , or at least after an hour or two, that it's gonna be more than 300GB

**Comment by C0D3D3V** [comment]
I think we could introduce a feature to check the size of most external files. Currently we don't check the size of external files in advance, so we can't tell how big the files (mostly videos from external sources) are before we start downloading them. Sometimes we can  not even tell the size if we start downloading the files, because some video formats or platforms do not tell how large the file is, so there is no way to estimate it in any way.  
For now we could tell the user somehow that the estimated size is only for moodle files and not external files. 

If someone is willing to implement this, we can have a discussion how I imagine this feature, and how it would be best to implement it. I personally would like to implement it, but have only little free time currently.

**Comment by masteoo** [comment]
@C0D3D3V good morning, sorry to bother, I opened an issue on EBSCO-dl, if you can help me. I am not a programmer:(
Thanks for your work!

**Comment by C0D3D3V** [comment]
@masteoo Don't worry I have seen your issue. There is no need to ping me in another issue :D But you can contact me at any time via mail.

**Comment by aaannndddyyy** [comment]
After downloading 357GB which supposedly are 87% of 407 GB I had to kill it.
It says 246 Failed. So I wanted to look in MoodleDL.log which ones failed, but that file is empty. Is it supposed to be empty ? 
Where can I learn which ones have failed so I can hunt them up manually and see if I can somehow get them ?

After freeing some space, can I just rerun 'moodle-dl' and it will resume where it left off without having to download again what's already downloaded, and fetching what's still missing and maybe retrying what has failed before ?

**Comment by aaannndddyyy** [comment]
Thank you

**Comment by aaannndddyyy** [comment]
If the size estimation is not really possible due to external sources, should I close this issue? Also, the percentage given there is clearly not 100 * downloaded / current_estimate.

**Comment by C0D3D3V** [comment]
We can leave the issue open for now, until we find a proper way to tell the user what is going on. It will help users that see that the estimation is increasing while downloading. Also we can discuss here about this topic and find solutions / ideas that we could implement.
